### PR TITLE
fix(figma.utils): error now exits the process gracefully"

### PR DESCRIPTION
**Pull request by DogaIster**
The error now exits the process gracefully

Fixes # (issue)
```
script/dist/commands/styles/utils/figma.utils.js:65
        throw new Error(msg || 'Assertion failed');
              ^

Error: fileId not found
    at assert 
```
now exits gracefully with a message for the user

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/cli/src/commands/styles/utils/figma.utils.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ export type FigmaData = {
 
 export const assert = <T>(x: T, msg?: string) => {
   if (!x) throw new Error(msg || 'Assertion failed');
+  console.log('Please make sure you entered a valid URL and token and try again');
```
Comment (sumitarora) [inline packages/cli/src/commands/styles/utils/figma.utils.ts:42]: should be `logger.error`

**Review by sumitarora — CHANGES_REQUESTED** [review]

**Review by rafRangler — CHANGES_REQUESTED** [review]
We need to figure out the best way to handle different errors

**Review by sumitarora — APPROVED** [review]

File: packages/cli/src/commands/styles/utils/figma.utils.ts [hunk 2]
```diff
@@ -38,14 +38,19 @@ export type FigmaData = {
...
+      logger.error(msg);
+    }
+    process.exit(1);
+  }
 };
 
 export function assertFigmaData(d: unknown): asserts d is FigmaData {
   const { fileId, nodeId, token } = (d as FigmaData) ?? {};
-  assert(typeof fileId === 'string', 'fileId not found');
+  assert(typeof fileId === 'string', 'Please put a valid URL');
```
Comment (sumitarora) [inline packages/cli/src/commands/styles/utils/figma.utils.ts:51]: `Please enter a valid URL`

File: packages/cli/src/commands/styles/utils/figma.utils.ts [hunk 3]
```diff
@@ -38,14 +38,19 @@ export type FigmaData = {
...
   assert(typeof nodeId === 'string', 'node-id not found');
-  assert(typeof token === 'string', 'token not found');
+  assert(typeof token === 'string', 'Please put a valid token');
```
Comment (sumitarora) [inline packages/cli/src/commands/styles/utils/figma.utils.ts:53]: `Please enter a valid token`

**Review by sumitarora — CHANGES_REQUESTED** [review]

**Review by sumitarora — APPROVED** [review]

**Review by rafRangler — APPROVED** [review]

File: packages/cli/src/commands/styles/lib/radius-styles.ts [hunk 4]
```diff
@@ -38,11 +36,11 @@ export const generateGlobalStyles = async ({
   assert(userToken !== 'none', 'Environment variable FIGMA_TOKEN is empty');
   assert(typeof url === 'string', 'Figma url must be provided');
   const renderTemplate = renderers[template];
-  const { getFileNode } = setup(userToken);
-  const { fileId, nodeId } = processFigmaUrl({ url, token: userToken });
+  
+  //const input = fs.existsSync(figmaFile) ? Promise.resolve(figmaFile): getFileNode(fileId,[nodeId]);
+
```
[the remark this answers is not in the document]
Comment (DogaIster) [reply packages/cli/src/commands/styles/lib/radius-styles.ts:41]: We shouldn't need this anymore either way but let's see

File: packages/cli/src/commands/styles/utils/figma.loader.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+export const loadFile = ({ url, token }: FigmaFileParams) => {
+
+  assert(url,'Figma URL Not Found, please run styles command again and provide the URL: ');
+  assert(token, 'Figma Token Not Found, please run styles command again and provide the token: ');
+
+  const { getFileNode } = setupFigma(token);
+  const { fileId, nodeId } = processFigmaUrl({ url, token });
+  assertFigmaData({ fileId, nodeId, token });
+
```
[the remark this answers is not in the document]
Comment (DogaIster) [reply packages/cli/src/commands/styles/utils/figma.loader.ts:25]: not sure about this, we don't own the backend, it throws a 403 AKA 'Not Authorized', that's not the error message I am giving. I rather not confuse the user

File: packages/cli/src/commands/styles/utils/figma.loader.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  });
+  return input;
+};
+
+export type FigmaData = {
+  fileId: string,
+  nodeId: string,
+  token: string,
+};
+
```
[the remark this answers is not in the document]
Comment (DogaIster) [reply packages/cli/src/commands/styles/utils/figma.loader.ts:38]: They are different things. So, the thing is they are both in the URL
`https://www.figma.com/file/<fieldID>/Radius-Design-Kit-V2(WIP)?node-id=<node ID>`
either way it means you messed up in the URL, node ID is not something we are asking for

**Pull request merged**